l1-contracts/gas_benchmark.md

```diff
@@ -1,20 +1,26 @@
-setupEpoch               | Avg     |      42110 (3.66)      |       61583 (5.35)      |      19473 (1.69)      |      46.24%
-setupEpoch               | Median  |      39724 (3.45)      |       39724 (3.45)      |          0 (0.00)      |       0.00%
-setupEpoch               | Max     |     112898 (9.80)      |      599729 (52.06)     |     486831 (42.26)     |     431.21%
+setupEpoch               | Min     |      37702 (3.27)      |       37702 (3.27)      |          0 (0.00)      |       0.00%
+setupEpoch               | Avg     |      41827 (3.63)      |       61569 (5.34)      |      19742 (1.71)      |      47.20%
+setupEpoch               | Median  |      39702 (3.45)      |       39702 (3.45)      |          0 (0.00)      |       0.00%
+setupEpoch               | Max     |     106360 (9.23)      |      599893 (52.07)     |     493533 (42.84)     |     464.02%
 setupEpoch               | # Calls |                    100 |                     100 |                      0 |       0.00%
 -------------------------+---------+------------------------+-------------------------+------------------------+-----------------
-submitEpochRootProof     | Min     |     867523 (75.31)     |      867523 (75.31)     |          0 (0.00)      |       0.00%
-submitEpochRootProof     | Avg     |     885223 (76.84)     |      885223 (76.84)     |          0 (0.00)      |       0.00%
-submitEpochRootProof     | Median  |     867607 (75.31)     |      867607 (75.31)     |          0 (0.00)      |       0.00%
-submitEpochRootProof     | Max     |     920541 (79.91)     |      920541 (79.91)     |          0 (0.00)      |       0.00%
+submitEpochRootProof     | Min     |     595697 (51.71)     |      595697 (51.71)     |          0 (0.00)      |       0.00%
+submitEpochRootProof     | Avg     |     627372 (54.46)     |      627372 (54.46)     |          0 (0.00)      |       0.00%
+submitEpochRootProof     | Median  |     595781 (51.72)     |      595781 (51.72)     |          0 (0.00)      |       0.00%
+submitEpochRootProof     | Max     |     690639 (59.95)     |      690639 (59.95)     |          0 (0.00)      |       0.00%
 submitEpochRootProof     | # Calls |                      3 |                       3 |                      0 |       0.00%
 -------------------------+---------+------------------------+-------------------------+------------------------+-----------------
+aggregate3               | Min     |          0 (0.00)      |      558189 (1550.53)   |     558189 (1550.53)   |       0.00%
+aggregate3               | Avg     |          0 (0.00)      |      576615 (1601.71)   |     576615 (1601.71)   |       0.00%
+aggregate3               | Median  |          0 (0.00)      |      569227 (1581.19)   |     569227 (1581.19)   |       0.00%
+aggregate3               | Max     |          0 (0.00)      |      638833 (1774.54)   |     638833 (1774.54)   |       0.00%
```

Comment:
What obvious things could it be that makes the max of this so much more expensive than the others? Oh, it is the first writes on all of it. I would suggest that you perform an initial signal before you start this, such that it is not a significant outlier. 

Without that combined with using the `medium` you would see insane savings in the setupepoch because the call now would be mainly no-ops.